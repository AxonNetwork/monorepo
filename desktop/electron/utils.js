const path = require('path')

module.exports.getAppPath = function() {
    try {
        return require('electron').app.getAppPath()
    } catch (err) {
        return require('electron').remote.app.getAppPath()
    }
}

module.exports.getEnv = function() {
    const appPath = module.exports.getAppPath()
    const binariesPath = path.join(appPath, '../desktop/build-resources/binaries')
    const env = Object.assign({}, process.env, { CONSCIENCE_APP_PATH: appPath, CONSCIENCE_BINARIES_PATH: binariesPath })
    env.PATH = binariesPath + ':' + env.PATH + ':/usr/local/bin:/usr/bin:/usr/sbin:/sbin'
    return env
}
