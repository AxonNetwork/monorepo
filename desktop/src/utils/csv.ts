export function parseCSV(contents: string){
	const lines = contents.split("\n")
	const data = lines.map(line => line.split(","))
	return data
}