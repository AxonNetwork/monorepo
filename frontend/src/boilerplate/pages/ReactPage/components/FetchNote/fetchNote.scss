@import 'sass/variables';
@import '../../reactPage';

.fetch-note-layout {
  @extend .react-block;
}

.fetch-note-title {
  @extend .react-block-title;
}

.fetch-note-collection {
  margin: 25px 0 0 0;
}

.fetch-note-error-panel {
  margin: 20px 0 0 0;
}

.fetch-note-filter-btn {
  width: 30%;
}

@media only screen and (max-width: $small-screen) {
  .fetch-note-filter-btn {
    width: 80%;
    max-width: 300px;
  }
}
