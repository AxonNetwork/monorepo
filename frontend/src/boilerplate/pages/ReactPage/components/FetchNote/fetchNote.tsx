import { List } from 'immutable';
import * as React from 'react';
import { InjectedI18nProps, InjectedTranslateProps, translate } from 'react-i18next';
import { connect } from 'react-redux';
import { AnyAction, Dispatch } from 'redux';

import { ADD_NOTE_REQUESTED, EDIT_NOTE_REQUESTED, FETCH_ALL_NOTES_REQUESTED, FETCH_NOTE_REQUESTED, REMOVE_NOTE_REQUESTED } from 'services/notes/constants';
import { INote } from 'services/notes/types';
import { IGlobalState } from 'types/global';
import i18ns from './i18n';

const styles = require('./fetchNote.scss');

// Component

interface IFetchNoteStateProps {
  notes: List<INote>;
  loading: boolean;
  error: string;
}

interface IFetchNoteDispatchProps {
  fetchAllNotes(): void;
  fetchNote(id: number): void;
  addNote(content: string): void;
  editNote(id: number, content: string): void;
  removeNote(id: number): void;
}

interface IFetchNoteProps extends IFetchNoteStateProps, IFetchNoteDispatchProps, InjectedI18nProps,  InjectedTranslateProps { }

export class FetchNote extends React.Component<IFetchNoteProps> {
  constructor(props: IFetchNoteProps) {
    super(props);
    this.loadI18ns();
  }

  public loadI18ns() {
    const { i18n } = this.props;
    Object.keys(i18ns).forEach((key) => {
      i18n.addResourceBundle(key, 'FetchNote', i18ns[key]);
    });
  }

  public fetchAllNotes = (e: React.MouseEvent<HTMLAnchorElement>) => {
    e.preventDefault();
    this.props.fetchAllNotes();
  }

  public render() {
    const { t } = this.props;
    const noteList = this.props.notes.map((note) =>
      (
        <li className='collection-item'>
          <div>Id: {note.id} &emsp; Content: {note.content}</div>
        </li>
      ),
    );
    const noteListCollection = <ul className={`collection ${styles['fetch-note-collection']}`}> {noteList} </ul>;
    const errorPanel = (
      <div className={`card-panel red ${styles['fetch-note-error-panel']}`}>
        <span className='white-text'>{this.props.error}</span>
      </div>
    );
    return (
      <div className={`center-align z-depth-2 ${styles['fetch-note-layout']}`}>
        <span className={styles['fetch-note-title']}>{t('async-calls')}</span>
        <div className='fetch-all-notes'>
          <a className={`btn waves-effect ${styles['fetch-note-filter-btn']}`} onClick={this.fetchAllNotes} role='button'>{t('fetch-all-notes')}</a>
        </div>
        {noteList.count() > 0 && noteListCollection}
        {this.props.error !== '' && errorPanel}
      </div>
    );
  }
}

// Container

const mapStateToProps = (state: IGlobalState): IFetchNoteStateProps => ({
  notes: state.notesState.notes,
  loading: state.notesState.loading,
  error: state.notesState.error,
});

const mapDispatchToProps = (dispatch: Dispatch<AnyAction>): IFetchNoteDispatchProps => ({
  fetchAllNotes: () => {
    dispatch({ type: FETCH_ALL_NOTES_REQUESTED, payload: {} });
  },
  fetchNote: (id: number) => {
    dispatch({ type: FETCH_NOTE_REQUESTED, payload: { id } });
  },
  addNote: (content: string) => {
    dispatch({ type: ADD_NOTE_REQUESTED, payload: { content } });
  },
  editNote: (id: number, content: string) => {
    dispatch({ type: EDIT_NOTE_REQUESTED, payload: { id, content } });
  },
  removeNote: (id: number) => {
    dispatch({ type: REMOVE_NOTE_REQUESTED, payload: { id } });
  },
});

export default connect(
  mapStateToProps,
  mapDispatchToProps,
)(translate('FetchNote')(FetchNote));
