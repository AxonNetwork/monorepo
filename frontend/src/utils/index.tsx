export const isProduction = process.env.NODE_ENV === 'production'

const exec = require('child_process').exec

export async function execCmd(cmd: string, cwd: string) {
    return new Promise((resolve, reject) => {
        exec(cmd, { encoding: 'utf8', cwd }, (error: Error|null, stdout: string, stderr: string) => {
            if (error) {
                reject(error)
            } else {
                resolve({ stdout, stderr })
            }
        })
    })
}
