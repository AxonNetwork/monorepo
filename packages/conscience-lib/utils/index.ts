export { default as autobind } from './autobind'
export { default as strToColor } from './strToColor'
export { default as schemes } from './codeColorSchemes'
export { default as getLanguage } from './getLanguage'
export { default as checkVisible } from './checkVisible'
export { default as nonCacheImg } from './nonCacheImg'
export { repoPageToString, stringToRepoPage } from './parseRepoPage'
export { orgPageToString, stringToOrgPage } from './parseOrgPage'
export { getConscienceURI } from './uri'
export { removeEmail, extractEmail } from './email'
export { parseCSV } from './csv'
export { isTextFile } from './fileTypes'
export { getLastVerifiedEvent, getLastVerifiedEventCommit, getLastVerifiedEventFile, getFirstVerifiedEvent } from './timeline'
