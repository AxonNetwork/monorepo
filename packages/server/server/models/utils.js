import { dynamo } from '../config/aws'

export function makeID() {
    let text = ''
    const possible = 'ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789'

    for (let i = 0; i < 10; i++) {
        text += possible.charAt(Math.floor(Math.random() * possible.length))
    }

    return text
}

export async function getAll(queryParams) {
    let items = []
    let ExclusiveStartKey

    do {
        const resp = await dynamo.queryAsync({
            ...queryParams,
            ExclusiveStartKey,
        })

        if (resp.Items) {
            items = items.concat(resp.Items)
        }

        ExclusiveStartKey = resp.LastEvaluatedKey
    } while (ExclusiveStartKey)

    return items
}
