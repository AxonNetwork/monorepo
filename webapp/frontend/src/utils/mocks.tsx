// Mock dispatch
export const dispatchMock = () => { return; };

// Mock translation
export const tMock = (k: string | string[]) => k;

// Mock i18n
export const i18nMock: any = {
  addResourceBundle: () => { return; },
};
