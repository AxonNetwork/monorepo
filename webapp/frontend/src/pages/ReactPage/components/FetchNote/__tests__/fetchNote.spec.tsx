import { List } from 'immutable';
import 'materialize-css';
import React from 'react';
import TestRenderer from 'react-test-renderer';

import { dispatchMock, i18nMock, tMock } from 'utils/mocks';
import { FetchNote } from '../fetchNote';

describe('FetchNote', () => {
  it('should renders correctly', () => {
    const renderer = TestRenderer.create(
      <FetchNote
        notes={List<any>()}
        loading={false}
        error={''}
        fetchAllNotes={dispatchMock}
        fetchNote={dispatchMock}
        addNote={dispatchMock}
        editNote={dispatchMock}
        removeNote={dispatchMock}
        t={tMock}
        tReady={true}
        i18n={i18nMock}
      />,
    );
    expect(renderer).toMatchSnapshot();
    renderer.unmount();
  });
});
