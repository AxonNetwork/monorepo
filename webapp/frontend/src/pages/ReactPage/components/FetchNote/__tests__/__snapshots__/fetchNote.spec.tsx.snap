// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FetchNote should renders correctly 1`] = `
<div
  className="center-align z-depth-2 fetch-note-layout"
>
  <span
    className="fetch-note-title"
  >
    async-calls
  </span>
  <div
    className="fetch-all-notes"
  >
    <a
      className="btn waves-effect fetch-note-filter-btn"
      onClick={[Function]}
      role="button"
    >
      fetch-all-notes
    </a>
  </div>
</div>
`;
